code/game/objects/items/melee/baton.dm

```diff
@@ -0,0 +1,656 @@
+		target.lastattackerckey = user.ckey
+		target.LAssailant = WEAKREF(user)
+		if(log_stun_attack)
+			log_combat(user, target, "stun attacked", src)
+	if(baton_effect(target, user, modifiers) && user)
+		set_batoned(target, user, cooldown)
+
+/obj/item/melee/baton/proc/baton_effect(mob/living/target, mob/living/user, modifiers, stun_override)
+	var/trait_check = HAS_TRAIT(target, TRAIT_STUNRESISTANCE)
+	if(iscyborg(target))
+		if(!affect_cyborg)
+			return FALSE
+		target.flash_act(affect_silicon = TRUE)
+		target.Paralyze((isnull(stun_override) ? stun_time_cyborg : stun_override) * (trait_check ? 0.1 : 1))
+		additional_effects_cyborg(target, user)
+	else
+		target.apply_damage(stamina_damage, STAMINA, BODY_ZONE_CHEST)
+		target.Knockdown((isnull(stun_override) ? knockdown_time : stun_override) * (trait_check ? 0.1 : 1))
+		additional_effects_non_cyborg(target, user)
+	return TRUE
+
+/// Description for trying to stun when still on cooldown.
+/obj/item/melee/baton/proc/get_wait_description()
+	return
+
+/// Default message for stunning a living, non-cyborg mob.
+/obj/item/melee/baton/proc/get_stun_description(mob/living/target, mob/living/user)
+	. = list()
+
+	.["visible"] =  "<span class ='danger'>[user] knocks [target] down with [src]!</span>"
+	.["local"] = "<span class ='userdanger'>[user] knocks you down with [src]!</span>"
+
+	return .
+
+/// Default message for stunning a cyborg.
+/obj/item/melee/baton/proc/get_cyborg_stun_description(mob/living/target, mob/living/user)
+	. = list()
+
+	.["visible"] = span_danger("[user] pulses [target]'s sensors with the baton!")
+	.["local"] = span_danger("You pulse [target]'s sensors with the baton!")
+
+	return .
+
+/// Default message for trying to stun a cyborg with a baton that can't stun cyborgs.
+/obj/item/melee/baton/proc/get_unga_dunga_cyborg_stun_description(mob/living/target, mob/living/user)
+	. = list()
+
+	.["visible"] = "<span class='danger'>[user] tries to knock down [target] with [src], and predictably fails!</span>" //look at this duuuuuude
+	.["local"] = "<span class='userdanger'>[target] tries to... knock you down with [src]?</span>" //look at the top of his head!
+
+	return .
+
+/// Contains any special effects that we apply to living, non-cyborg mobs we stun. Does not include applying a knockdown, dealing stamina damage, etc.
+/obj/item/melee/baton/proc/additional_effects_non_cyborg(mob/living/target, mob/living/user)
+	return
+
+/// Contains any special effects that we apply to cyborgs we stun. Does not include flashing the cyborg's screen, hardstunning them, etc.
+/obj/item/melee/baton/proc/additional_effects_cyborg(mob/living/target, mob/living/user)
+	return
+
+/obj/item/melee/baton/proc/set_batoned(mob/living/target, mob/living/user, cooldown)
+	if(!cooldown)
+		return
+	var/user_ref = REF(user) // avoids harddels.
+	ADD_TRAIT(target, TRAIT_IWASBATONED, user_ref)
+	addtimer(TRAIT_CALLBACK_REMOVE(target, TRAIT_IWASBATONED, user_ref), cooldown)
```

Comment:
if this batong gets removed, this timer also gets removed.

Reply:
No, `TRAIT_CALLBACK_REMOVE` and `TRAIT_CALLBACK_ADD` is a global proc callback. The timer won't be removed.